libsolidity/ast/Types.cpp

```diff
@@ -2957,12 +2957,30 @@ FunctionTypePointer FunctionType::interfaceFunctionType() const
 	);
 }
 
-MemberList::MemberMap FunctionType::nativeMembers(ContractDefinition const*) const
+MemberList::MemberMap FunctionType::nativeMembers(ContractDefinition const* _scope) const
 {
 	switch (m_kind)
 	{
 	case Kind::Declaration:
-		return {{"selector", TypeProvider::fixedBytes(4)}};
+		if (declaration().isPartOfExternalInterface())
+			return {{"selector", TypeProvider::fixedBytes(4)}};
+		else
+			return MemberList::MemberMap();
+	case Kind::Internal:
+		if (
+			auto const* functionDefinition = dynamic_cast<FunctionDefinition const*>(m_declaration);
+			functionDefinition &&
+			_scope &&
+			functionDefinition->annotation().contract &&
+			_scope != functionDefinition->annotation().contract &&
```

Comment:
Maybe this could warrant a different function kind to be honest...

Reply:
Like ``Public``? Maybe... not sure if I should do that now, though... I already spent way too much time on this - and I'm not particularly happy with it myself yet, but I'm not sure how much more time we should spend on refactoring this right now...
The problem is that I'd probably need two more kinds ``PublicWithInternalCallingContext`` and ``PublicWithExternalCallingContext``... in general it would probably be good to decouple the function type or at least the function type kind from the calling context a bit... but that would require major code changes...